build(refactor): maestro builds with fastlane instead of local

gkartalis opened the pull request:
This PR resolves [] <!-- eg [PROJECT-XXXX] -->

### Description

Updates `build-maestro-ios.yml` and `build-maestro-android.yml` to use fastlane for bumping the versions properly and for consistency.



<!-- Info, implementation, how to get there, before & after screenshots & videos, follow-up work, etc -->

<!--  Please include screenshots or videos for visual changes, at least on Android -->
<!-- Screenshots template:
| Platform | Before | After |
|---|---|---|
| Android | <img src="xxx" width="400" /> | <img src="xxx" width="400" /> |
| iOS | <img src="xxx" width="400" /> | <img src="xxx" width="400" /> |
-->
<!-- Videos template:
| Platform | Before | After |
|---|---|---|
| Android | <video src="xxx" width="400" /> | <video src="xxx" width="400" /> |
| iOS | <video src="xxx" width="400" /> | <video src="xxx" width="400" /> |
-->

### PR Checklist

- [ ] I have tested my changes on the following platforms:
  - [ ] **Android**.
  - [ ] **iOS**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos** at least on **Android**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

### Changelog updates

<!-- 📝 Please fill out at least one of these sections. -->
<!-- ⓘ 'User-facing' changes will be published as release notes. -->
<!-- ⌫ Feel free to remove sections that don't apply. -->
<!-- • Write a markdown list or just a single paragraph, but stick to plain text. -->
<!-- 📖 eg. `Enable lotsByFollowedArtists` or `Fix phone input misalignment`. -->
<!-- 🤷‍♂️ Replace this entire block with the hashtag `#nochangelog` to avoid updating the changelog. -->
<!-- ⚠️ Prefix with `[NEEDS EXTERNAL QA]` if a change requires external QA -->

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- maestro builds with fastlane instead of local

<!-- end_changelog_updates -->

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

ArtsyOpenSource commented:
<!--
  0 failure: 
  0 warning: 
  
  1 markdown notices
  DangerID: danger-id-Danger;
-->



### This PR contains the following changes:

- Dev changes (wip wip - gkartalis)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 63fa9790d386d5bd0dea351bb8aec286f1296fbe
</p>

gkartalis commented on .github/workflows/build-maestro-android.yml:
```diff
@@ -4,6 +4,9 @@ on:
   schedule:
     - cron: "0 6 * * *" # Runs daily at 6:00 AM UTC
   workflow_dispatch: # Allows manual trigger
+  push:
+    branches:
+      - gkartalis/PHIRE-2090
```
to be reverted - was added in order to run it against this PR

gkartalis commented on .github/workflows/build-maestro-ios.yml:
```diff
@@ -4,10 +4,13 @@ on:
     - cron: "0 6 * * *" # Runs daily at 6:00 AM UTC
   workflow_dispatch: # Allows manual trigger
+  push:
+    branches:
+      - gkartalis/PHIRE-2090
```
to be removed - was added for testing changes against this PR

gkartalis commented on fastlane/Fastfile:
```diff
@@ -307,24 +308,44 @@ lane :ship_beta_android do |options|
-      sentry_release_name: sentry_release_name,
-      dist_version: "#{vcode}",
-      platform: 'android'
+  if maestro
```
Should we keep the maestro here as a flag in order to reuse the `ship_beta_android` lane for everything or should we keep it as a standalone lane as we do further down with iOS?

brainbicycle replied:
i think we should separate it to keep the beta lanes clean as we do on iOS

gkartalis commented on fastlane/Fastfile:
```diff
@@ -560,3 +581,74 @@ error do |lane, exception|
 end
+
+desc 'Build iOS app and upload to S3 for Maestro testing'
+lane :build_maestro_ios do
```
this is the alternative that we have - to keep the maestro build on a separate lane - open to feedback

gkartalis commented:
addressed all the comments - ready for another review - also running the builds once more to make sure I didn't break anything with the refactors cc @brainbicycle
